p2p/peer_tracker.go

```diff
@@ -105,6 +106,25 @@ func (p *peerTracker) track() {
 	}
 }
 
+// getPeers returns the tracker's currently tracked peers.
+func (p *peerTracker) getPeers(amount int) ([]peer.ID, error) {
+	p.peerLk.RLock()
+	defer p.peerLk.RUnlock()
+
+	if len(p.trackedPeers) < amount {
+		return nil, fmt.Errorf("not enough peers in tracker: requested %d, have %d", amount, len(p.trackedPeers))
```

Comment:
So if we have <4 tracked peers we will use trusted peers as per logic in Head, but note that tracked peers contain all the trusted.

So in case we have 2 trusted and 1 tracked, we will only use trusted because of this error.

We trustedHead is set, we should use all the tracked peers even if that's less than 4. We can log out such event as well.

Reply:
https://github.com/celestiaorg/go-header/pull/53/commits/d3dae259d47e979d41e1ea4d939bb914293c5896

This will mitigate the concern that peer tracker will be < `numUntrustedHeadRequests` on start.

Reply:
> Also, with peer dumper, it's unlikely that the case will happen that tracker is empty or sub 4 peers.

I agree that that's the rare case, but it's more like the code is doing unnecessary checks and its written like peerTracker and trustedPeers are fully separated things, while in fact peerTracker contains all the trusted.

Reply:
@renaynay